### DISCUSSION TITLE
Interactive computational notebooks via emacs-jupyter

### DISCUSSION THREAD

**Discussion by dangirsh** [emacs-ng/emacs-ng#474] [General]
Excited to find this project today!

I wanted to bring awareness to the excellent [emacs-jupyter](https://github.com/nnicandro/emacs-jupyter) package, which I use for interactive/literate programming in Julia and Python (usually on remote HPC resources). A powerful workflow is generating a test environment (e.g by running a notebook via the org-mode client), then using the "associate buffer to kernel" feature to allow evaluating library code (e.g. a function being developed / debugged) within that context, with inline results. It really feels like the "professional" version of the standard browser-based Jupyter notebook.

One of the biggest missing features is support for interactive browser-based elements like animations, sliders, maps, interactive plots, iframes, etc... Right now, there's [experimental widget support](https://github.com/nnicandro/emacs-jupyter#building-the-widget-support-experimental) (which I never got to work) which relies on an external browser. There's also [this work](https://github.com/nnicandro/emacs-jupyter/pull/198), where a screenshot of a browser session is retrieved. 

It seems like this project may have the potential of "finishing" emacs-jupyter by enabling these interactive browser-based elements. Combined with the existing org-mode client, I think this would be a seriously supercharged interactive computing environment.

I'm curious to hear any thoughts about the feasibility of this integration. If anyone sees a clear roadmap towards this end, a summary of the remaining work would be highly appreciated.

**Comment by DavidDeSimone** [comment]
I've been meaning to propose a feature that shows off the capabilities we get using one of the more popular deno plugins (https://github.com/webview/webview_deno), and I think this could be it. 

I've demo'd using webview and it works well (the only cavet being you need to download the shared library prior to launch), but we can communicate between the emacs JS/elisp and the WebViews JS layer.
